doc/SEMANTICS.md

```diff
@@ -88,7 +88,12 @@ Renaming files and directories (`rename`, `renameat`) is not currently supported
 
 File deletion (`unlink`) is not currently supported, but will be [in the future](https://github.com/awslabs/mountpoint-s3/issues/78).
 
-Empty directory removal (`rmdir`) is not supported.
+Empty directory removal (`rmdir`) is not currently supported, but will be [in the future](https://github.com/awslabs/mountpoint-s3/issues/194). The following semantics are proposed:
+
+* `rmdir` will only delete empty directories created by `mkdir`.
+* `rmdir` will fail on directories backed on S3 by a directory marker (i.e. zero-byte object with `<directory-name>/` key).
+* Once, the files are commited to S3 bucket for a directory, `rmdir` will not affect those directories any further, even if they are created by `mkdir`.
```

Comment:
Once the files are commited to S3 bucket, the status of the directory becomes remote. So, after we remove all the objects from the directory, the directory itself will get removed as it will not be there in the remote view.

Reply:
I see. I don't really understand what this is trying to tell me then.

What we're really saying is that when a file is committed, the directory becomes implicit due to the presence of remote files. As soon as Mountpoint observes that those files no longer exist, the directory will no longer exist in Mountpoint.

Reply:
Maybe something like...

```markdown
* As soon as a file is committed to the S3 bucket by Mountpoint,
  the directory will be considered to exist implicitly.
  If Mountpoint later observes that there are no files existing for that directory in S3, 
  Mountpoint will consider the directory to have been deleted.
```